sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceManager.scala

```diff
@@ -75,3 +86,136 @@ class DataSourceManager {
     dataSourceBuilders.containsKey(normalize(name))
   }
 }
+
+/**
+ * Data Source V1 wrapper for Python Data Source.
+ */
+object PythonDataSourceCodeGenerator extends Logging {
+  def generateClass(shortName: String): Class[_] = {
+    val ctx = newCodeGenContext()
+
+    val codeBody = s"""
+      private ${classOf[String].getName} source = "$shortName";
```

Comment:
This is very interesting. We basically dynamically generate a DSv1 wrapper class when registering a Python data source. I find it a bit difficult to read the code body. Just curious, can we replace `classOf[String].getName` to `String` in this case?

Reply:
We could do but we should write `java.lang.String` .. that is actually more and less same I believe.